### ISSUE TITLE
How do delete Attachement folder?

### ISSUE THREAD

**Issue by mike4001** [bbernhard/signal-cli-rest-api#635]
Hi,

During my Home Assistant Backups I noticed that my Signal Addon Backup is using almost 1,3 GB of space.

Looking into the compressed backup file I am seeing that this is mostly in the attachement folder (images and videos sent into a group where the Signal Number is a member of).

Is it possible to delete these files? If yes, how? I don' really have access to the File structure of Home Assistant Addons.
Is it possible that they are not downloaded in the first place?

Thanks

**Comment by mike4001** [comment]
OK, I found the attatchment folder under

\\Home-Assistant-IP\addon_configs\1315902c_signal_messenger\attachments

So there I can just delete the files?

Second question remains. Is it possible to not get the folder this big in the first place?

**Comment by bbernhard** [comment]
If you are using the normal/native mode, there is the ìgnore_attachments` parameter (https://bbernhard.github.io/signal-cli-rest-api/#/Messages/get_v1_receive__number_)

**Comment by mike4001** [comment]
Thank you for your answer but I do not see any ìgnore_attachment command in your posted link?

**Comment by kamuridesu** [comment]
@mike4001 Here
![image](https://github.com/user-attachments/assets/4fec2793-5fb4-4763-8760-90d5a73ebd6f)

**Comment by LordShadowen** [comment]
I have a similar issue as OP - large attachments folder, which I don't really need or want. 

If I understood the comments correctly, if I was using the normal or native mode, I would be in charge of somehow calling the `receive` API, so I could control if I wanted attachments or not.

But I'm using the `json-rpc` mode, so messages are just automatically being received (in real time) and stored without me doing anything (correct?)

And it seems that in this case, there's no option to specify if I want the attachments or not - they are just downloaded and stored automatically. 

So - if all of these assumptions are correct - is there any workaround or trick I could use to disable this? If not, I guess then I have a "feature request" to add such an option :)

**Comment by bbernhard** [comment]
> I have a similar issue as OP - large attachments folder, which I don't really need or want.
> 
> If I understood the comments correctly, if I was using the normal or native mode, I would be in charge of somehow calling the `receive` API, so I could control if I wanted attachments or not.
> 
> But I'm using the `json-rpc` mode, so messages are just automatically being received (in real time) and stored without me doing anything (correct?)
> 
> And it seems that in this case, there's no option to specify if I want the attachments or not - they are just downloaded and stored automatically.
> 
> So - if all of these assumptions are correct - is there any workaround or trick I could use to disable this? If not, I guess then I have a "feature request" to add such an option :)

That's correct. Unfortunately, the upstream signal-cli currently does not provide an option for excluding the attachment download in jsonRpc mode. At least I couldn't find anything in the [documentation](https://github.com/AsamK/signal-cli/blob/master/man/signal-cli-jsonrpc.5.adoc). 

I guess what I could do is add a configurable background task, which automatically cleans up the attachment folder. But not sure if this is a good idea - feels like a workaround :/

In any case, what you could do right now is to use the  `/v1/attachments` endpoints to list and delete the attachments yourself.
